doc/user_guide/signal2d.rst

```diff
@@ -115,3 +115,136 @@ while the offset is determined by the `offset` parameter. The fulcrum of the
 linear ramp is at the origin and the slopes are given in units of the axis
 with the according scale taken into account. Both are available via the
 :py:class:`~.axes.AxesManager` of the signal.
+
+
+Peak finding
+------------
+
+.. versionadded:: 1.6
+
+The :py:meth:`~._signals.signal2d.Signal2D.find_peaks` method provides access
+to a number of algorithms for that achieve peak finding in two dimensional
+signals. The methods available are as follows:
+
+Maximum based peak finder
+^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. code-block:: python
+
+    >>> s.find_peaks(method='local_max')
+    >>> s.find_peaks(method='max')
+    >>> s.find_peaks(method='minmax')
+
+These methods are searching peaks using maximum (and mimimum) values in the
+image. There all have a ``distance`` parameter to set the minimum distance
+between the peaks.
+- the ``'local_max'`` method uses the :py:func:`skimage.feature.peak_local_max`
+  function (``distance`` and ``threshold`` parameters are mapped to
+  ``min_distance`` and ``threshold_abs``, respectively).
+- the ``'max'`` method uses the
+  :py:func:`~.utils.peakfinders2D.find_peaks_max` function to search
+  for peaks higher than ``alpha * sigma``, where ``alpha`` is parameters and
+  ``sigma`` is the standard deviation of the image. It also has a ``distance``
+  parameters to set the minimum distance between peaks.
+- the ``'minmax'`` method uses the
+  :py:func:`~.utils.peakfinders2D.find_peaks_minmax` function to locate
+  the positive peaks in an image by comparing maximum and minimum filtered
+  images. Its ``threshold`` parameter defines the minimum difference between
+  the maximum and minimum filtered images.
+
+Zaeferrer peak finder
+^^^^^^^^^^^^^^^^^^^^^
+
+.. code-block:: python
+
+    >>> s.find_peaks(method='zaefferer')
+
+This algorithm was developed by Zaefferer [1]_ and the
+implementation here is after the description of the algorithm in the Ph.D.
+thesis of Thomas A. White. It is based on a gradient threshold followed by a
+local maximum search within a square window, which is moved until it is
+centered on the brightest point, which is taken as a peak if it is within a
+certain distance of the starting point. It uses the
+:py:func:`~.utils.peakfinders2D.find_peaks_zaefferer` function, which can
+take the ``grad_threshold``, ``window_size`` and ``distance_cutoff`` as
+parameters. See the :py:func:`~.utils.peakfinders2D.find_peaks_zaefferer`
+function documentation for more details.
+
+Ball statistical peak finder
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. code-block:: python
+
+    >>> s.find_peaks(method='stat')
+
+Developed by Gordon Ball, and described in the Ph.D. thesis of Thomas A.
+White, this method is based on finding points which have a statistically
+higher value than the surrounding areas, then iterating between smoothing and
+binarising until the number of peaks has converged. This method is slow, but
```

Comment:
It _can be slow_. Perhaps mention that if you have `numba` installed, it will automatically run much faster?